pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -282,11 +281,26 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 			for _, pr := range pullRequests {
 				createTime := pr.CreatedAt
 				prAgeUpperBound := now.AsTime()
+				approvalAgeUpperBound := now.AsTime()
+				deploymentAgeUpperBound := now.AsTime()
```

Comment:
can we not set `var ...UpperBound time.Time` instead of setting it to right now & then replacing it if a conditional is met?

Reply:
My thought process was if the age of whatever the metric is is either from the creation time until now or until those other conditions have been met which is why I structured it this way.  Not sure where else they'd be set either since this is the only place they're used

Reply:
for pr time yes, but for deployment time & merged time, if there's no deployment or isn't merged than the metric wouldn't be created.

Reply:
Was going to ask about this actually.  So open pr age is the only thing we care to continuously output here, everything else is just important after we know it exists?